pkg/provider/aws/provider.go

```diff
@@ -78,17 +82,35 @@ func retrieveVPCID() (string, error) {
 	if vpcID != "" {
 		return vpcID, nil
 	}
-	mac, err := retrieveInstanceNetworkInterfaceMacAddress()
+
+	client := &http.Client{}
+	req, _ := http.NewRequest(http.MethodPut, tokenEndpoint, nil)
+	req.Header.Set("X-aws-ec2-metadata-token-ttl-seconds", "21600")
+	res, err := client.Do(req)
 	if err != nil {
 		return "", err
 	}
```

Comment:
The defer res.Body.Close() should be executed just after executing the request, in case io.ReadAll fails. 
```suggestion
	req.Header.Set("X-aws-ec2-metadata-token-ttl-seconds", "21600")
	res, err := client.Do(req)
	if err != nil {
		return "", err
	}
	defer res.Body.Close()
```